frontend/src/views/WheresMyGene/components/HeatMap/components/Chart/components/Chart/hooks/utils.ts

```diff
@@ -0,0 +1,196 @@
+import {
+  EChartsOption,
+  DatasetComponentOption,
+  ScatterSeriesOption,
+} from "echarts";
+import { ViewId } from "src/views/WheresMyGene/common/types";
+
+export interface CreateChartOptionsProps {
+  /**
+   * The data array to be visualized
+   * The data point object shape can be whatever you like, but it must be consistent with the `encode` option
+   * For example, if the data point shape is:
+   * {
+   *   geneIndex: 0,
+   *   cellTypeIndex: 0,
+   *   percentage: 0.5
+   * }
+   * and you want geneIndex to be encoded to x axis and cellTypeIndex to be encoded to y axis, then make sure your encode option is:
+   * encode: {
+   *   x: 'geneIndex',
+   *   y: 'cellTypeIndex'
+   * }
+   */
+  data: DatasetComponentOption["source"];
```

Comment:
retyped this to use echart's own type (`atasetComponentOption["source"]`) with an example of data shape we typically use in heatmap as comment!

Originally I wanted to use a generic type to convey that a user can pass whatever object they want for this, but I figured the message comes across better by explicitly saying so in the comment and referring to the echarts' source of truth directly